src/aptos_container.rs

```diff
@@ -237,27 +408,147 @@ impl AptosContainer {
+    ///     let _ = aptos_containe.run(2, |accounts| {
+    ///             Box::pin(async move {
+    ///                 let aptos_container = AptosContainer::init().await.unwrap();
+    ///                 let accounts = aptos_container.get_initiated_accounts().await.unwrap();
+    ///                 let module_account_private_key = accounts.first().unwrap();
+    ///                 let module_account_address = get_account_address(module_account_private_key);
+    ///                 let mut named_addresses = HashMap::new();
+    ///                 named_addresses.insert("verifier_addr".to_string(), module_account_address);
+    ///                 aptos_container
+    ///                     .upload_contract(
+    ///                         "./contract-samples/sample1",
+    ///                         module_account_private_key,
+    ///                         &named_addresses,
+    ///                         None,
+    ///                         false,
+    ///                     )
+    ///                     .await
+    ///                     .unwrap();
+    ///                 Ok(())
+    ///             })
+    ///     });
+    /// }
+    /// ```
+    pub async fn run(
+        &self,
+        number_of_accounts: usize,
+        callback: impl FnOnce(Vec<String>) -> Pin<Box<dyn Future<Output = Result<()>>>>,
+    ) -> Result<()> {
+        // Ensure that accounts are initialized, if not already done.
+        self.lazy_init_accounts().await?;
+
+        // Determine whether to use overridden accounts or to receive them via the channel.
+        let accounts = match &self.override_accounts {
+            // If override_accounts is Some, clone the provided accounts.
+            Some(accounts) => accounts.clone(),
+            // Otherwise, receive the accounts from the accounts_channel_rx.
+            None => {
+                // TODO: check received messages size
+                let mut result = vec![];
+                // Lock the accounts_channel_rx to ensure exclusive access and receive accounts.
+                self.accounts_channel_rx
+                    .lock()
+                    .await
+                    .as_mut()
+                    .unwrap()
+                    .recv_many(&mut result, number_of_accounts)
+                    .await;
+                result
+            }
+        };
+
+        // Invoke the provided callback with the received or overridden accounts.
+        let result = callback(accounts.clone()).await;
+
+        if self.override_accounts.is_none() {
+            let guard = self.accounts_channel_tx.read().await;
+            for account in accounts {
+                guard.as_ref().unwrap().send(account).await?;
+            }
+        }
+        result
+    }
+
+    /// Executes a script located within the specified directory.
+    ///
+    /// # Parameters
+    /// - `local_dir`: The directory path containing the scripts to be executed.
```

Comment:
The directory path pointing to a Move package where its script will be executed

Reply:
done